linode/helper/framework_conversion.go

```diff
@@ -43,6 +44,22 @@ func StringSliceToFramework(val []string) []types.String {
 	return result
 }
 
+// StringSliceToFramework converts the given string slice
+// into a framework-compatible slice of attr.Value.
+func StringSliceToFrameworkGeneric(val []string) []attr.Value {
```

Comment:
I'm thinking if it's better to return `[]types.String` instead of `[]attr.Value` to be more specific.

Reply:
The function who returns `[]types.String` is [right below this one](https://github.com/linode/terraform-provider-linode/pull/1280/files#diff-6715b511a073485e1e06abd89d7985c9bd214181c6064cbb4bfb9dbd9a0a14c6R65)..

Casting to `[]attr.Value` is intentional because we sometimes need to convert `[]string` to `[]attr.Value` because `[]types.String` can't be passed as `[]attr.Value` in golang even if `types.String` implements `attr.Value`.

Reply:
The below function `FrameworkSliceToString` actually returns `[]string` which converts the framework type to the go type. I thought this function you're building is doing the revert way.

>  []types.String can't be passed as []attr.Value in Golang even if types.String implements attr.Value

I think it can work if the caller wants a `[]types.String` specifically. Just reviewed the way you called this function is from `KeepOrUpdateSet(...)`, which takes in an `[]attr.Value`. This should be the reason that this function is returning like this way.

Reply:
Ah, sorry, it was [the function above it](https://github.com/linode/terraform-provider-linode/pull/1280/files#diff-6715b511a073485e1e06abd89d7985c9bd214181c6064cbb4bfb9dbd9a0a14c6L32), not the one below it..